### DISCUSSION TITLE
Allow Custom CSV File Path in ImportAction

### DISCUSSION THREAD

**Discussion by misterneo** [filamentphp/filament#14257] [Ideas]
Hello @danharrin,

I hope you're doing well! Before proceeding with the implementation, I wanted to check whether the following feature would be suitable.

The idea is to allow specifying a custom CSV file path in the ImportAction class. This would be particularly useful for cases where specific columns are required that aren't part of the auto-generated file. I believe this minor addition could greatly improve flexibility for v3 users without requiring substantial changes.

Given that new features are currently closed for v3, I'd love your thoughts on whether this could be an exception or if it would be better suited for the v4.x branch.

Thanks for your guidance!

**Comment by danharrin** [comment]
Hi, sorry if I missed this, but what would the file actually be used for? Is it the "download example" button?

**Reply by misterneo** [reply]
Yes the `Download example CSV file` link button

**Reply by danharrin** [reply]
You can submit a PR to v4 if you wish, but it might be more flexible to allow any array of CSV content to be passed instead of just a file path. You could use the file path to construct an array instead.

**Reply by misterneo** [reply]
I was thinking something as quick and simple as this:

![Screenshot 2024-09-16 at 12 41 44](https://github.com/user-attachments/assets/b503b085-cc87-452f-b3f9-a97d3b6bacdf)

I’m currently working on a project that could benefit from this feature. At the moment, I have to instruct users not to use the 'Download CSV Example' button. Instead, I provide them with a custom CSV file directly. As a temporary workaround, I’m considering hiding the button using CSS.

**Reply by danharrin** [reply]
It's not as flexible as if we just accepted an array of data that could be sourced from anywhere

**Reply by misterneo** [reply]
I really appreciate your suggestion that would definitely make things more flexible.

I actually found a workaround by calling the `registerModalActions` function on the `ImportAction` and providing the path to a custom file, which works well for now. If I get a chance to dive into it later, I’ll keep your idea in mind for a potential PR.

Thanks for your help!
